packages/lib/src/vis/tiles/TiledHeatmapLayer.tsx

```diff
@@ -0,0 +1,61 @@
+import { Suspense } from 'react';
+import { LinearFilter, NearestFilter, Vector2 } from 'three';
+
+import { useVisibleDomains } from '../hooks';
+import Tile from './Tile';
+import type { TilesApi } from './api';
+import type { ColorMapAttributes } from './models';
+import { getTilesOffset, tileToCenterSquareDistance } from './utils';
+
+interface Props extends ColorMapAttributes {
+  api: TilesApi;
+  lod: number;
+}
+
+function TiledHeatmapLayer(props: Props) {
+  const { api, lod, ...colorMapAttrs } = props;
+
+  const scale = 2 ** lod;
```

Comment:
This is coupled on how you compute `levelSizes` in `getLevelsSize` **but is in no way enforced**. This is quite problematic as the API and this component could end up computing `scale` differently.

One solution would be to make the scale computation a method of the `api`. Perhaps do you have a better idea ?

Reply:
Indeed.
By moving group's scale and position in `TiledHeatmapLayer`, it is possible to use the right scale here computed from the level's size.